### PR TITLE
Add BackupTarget unavailable reason and HTTP endpoint

### PR DESCRIPTION
**Pull request by jenting**
#### Issues

https://github.com/longhorn/longhorn/issues/2887

### CODE REVIEW

File: api/model.go [hunk 1]
```diff
@@ -1072,6 +1078,28 @@ func toSnapshotCollection(ss map[string]*types.Snapshot) *client.GenericCollecti
...
+			Id:    bt.Name,
+			Type:  "backupTarget",
+			Links: map[string]string{},
+		},
+		BackupTarget: engineapi.BackupTarget{
+			BackupTargetURL:  bt.Spec.BackupTargetURL,
+			CredentialSecret: bt.Spec.CredentialSecret,
+			PollInterval:     bt.Spec.PollInterval.Duration.String(),
+			Available:        bt.Status.Available,
+			Message:          bt.Status.Conditions[types.BackupTargetConditionTypeUnavailable].Message,
```
Comment (innobead) [inline api/model.go:1097]: If the condition unavailable, have a default message to avoid null pointer exception.

File: manager/engine.go [hunk 2]
```diff
@@ -225,6 +225,23 @@ func (m *VolumeManager) GetEngineClient(volumeName string) (client engineapi.Eng
 	})
 }
 
+func (m *VolumeManager) ListBackupTargetsSorted() ([]*longhorn.BackupTarget, error) {
+	backupTargetMap, err := m.ds.ListBackupTargets()
+	if err != nil {
+		return []*longhorn.BackupTarget{}, err
+	}
+
+	backupTargets := make([]*longhorn.BackupTarget, len(backupTargetMap))
```
Comment (innobead) [inline manager/engine.go:234]: Suggest to move after the below if-err block, so just use it when needed.

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just few comments needed to resolve.

File: api/model.go (hunk 1 above)
Comment (jenting) [reply api/model.go:1097]: I think NPE won't happens https://play.golang.org/p/0gO7qqleBxo

**Review by innobead — APPROVED** [review]

**Pull request merged**